packages/aztec.js/src/proof/joinSplit/index.js

```diff
@@ -127,6 +128,32 @@ class JoinSplitProof extends Proof {
         );
     }
 
+    /**
+     * Construct the EIP712 signatures, giving permission for notes to be spent
+     * @param {string} validatorAddress Ethereum address of the join-split validator contract
+     * @param {string[]} aztecAccounts mapping between owners and private keys
+     * @returns {string} array of signatures
+     */
+    constructSignatures(validatorAddress, inputNoteOwners) {
```

Comment:
I suggest rewriting this as a mapping instead of an array. Find an example here: https://github.com/AztecProtocol/AZTEC/blob/0d5af182ca4d7698654ffbfd348f02a555464d94/packages/aztec.js/src/proof/joinSplit/index.js#L131

You can merge over `this.inputNotes` instead of `inputNoteOwners`.

Aslo, make sure the comments are in sync with the parameter names.

Reply:
As discussed, leaving for now